### PR TITLE
MDK-3871: SpotCLI EKS

### PR DESCRIPTION
**Pull request by arnoldyahad**
# MDK-3871 EKS actions on SpotCLI

this PR allows actions of EKS to be performed on spotCLI:

image

there are 3 new commands:

### Replicas
get current number of replicas:
image

### Annotations
image

## Pause
this command puts a pause-replicas annotation:
image

override the annotation and scale up/down while being paused using -o flag:

image

remove the annotation(unpause) using -u flag:
image

### CODE REVIEW

**Review by liorfranko — COMMENTED** [review]
Regarding the CLI convention.
It looks like pretty different from the existing spotcli.
Why not try to stick to the convention?
The pause command can be split to `suspend/unsuspend` and maybe add `set` instead of `-o`
I would ask for a short review from Guni or someone else from Delivery just regarding the CLI and write it from their perspective and not ours.

File: spotcli/k8s.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+def connectToApi():
+    eks_client = dynamic.DynamicClient(
+        api_client.ApiClient(configuration=config.load_kube_config())
+    )
+    return(eks_client)
+
+@timeout_decorator.timeout(10,timeout_exception=TimeoutError)
+def get_deployment_replicas(namespace, deployment):
+    eks_client = connectToApi()
+    api = eks_client.resources.get(api_version="argoproj.io/v1alpha1", kind="Rollout")
```
[the remark this answers is not in the document]
Comment (arnoldyahad) [reply spotcli/k8s.py]: fixed this issue

File: spotcli/k8s.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    try:
+        deployment_created = api.get(name=deployment, namespace=namespace)
+    except ApiException:
```
[the remark this answers is not in the document]
Comment (arnoldyahad) [reply spotcli/k8s.py:26]: it prints a huge huge json that is not really readable.
when there will be something that is not API and not timeout - we will get it

File: spotcli/k8s.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        return(None)
+    stats['currentReplicas'] = keda_hpa.status.currentReplicas
+    stats['desiredReplicas'] = keda_hpa.status.desiredReplicas
+    stats['maxReplicas'] = keda_hpa.spec.maxReplicas
+    stats['minReplicas'] = keda_hpa.spec.minReplicas
+    return(yaml.dump(stats))
+
+@timeout_decorator.timeout(10,timeout_exception=TimeoutError)
+def get_scaled_object(namespace, deployment):
+    eks_client = connectToApi()
```
[the remark this answers is not in the document]
Comment (arnoldyahad) [reply spotcli/k8s.py:40]: yes

File: spotcli/cli.py [hunk 4]
```diff
@@ -316,14 +324,193 @@ def scale(kind: str, group: str, amount: str, auto_approve: bool) -> None:
...
+)
+@click.option(
+    "-d",
+    "--deployment",
+    type=click.STRING,
+    default=False,
+    is_flag=True,
+    help="Choose a Deployment",
+)
+def stats(namespace: str, deployment: str) -> None:
```
[the remark this answers is not in the document]
Comment (arnoldyahad) [reply spotcli/cli.py:348]: replicas is being used as a flag so its stops me from using it again, weird click limitation
but i thought to do "stats" so in the future we will augment this command to show more info

**Review by abirsigron — APPROVED** [review]

**Pull request merged**